wpilibj/src/athena/java/edu/wpi/first/wpilibj/Counter.java

```diff
@@ -541,8 +540,10 @@ public void setDistancePerPulse(double distancePerPulse) {
   public void setPIDSourceType(PIDSourceType pidSource) {
     if (pidSource == null) {
       throw new NullPointerException("PID Source Parameter given was null");
+    } else if (pidSource == PIDSourceType.kDisplacement || pidSource == PIDSourceType.kRate) {
```

Comment:
This logic is backwards (will always throw), should be `pidSource != kDisplacement && pidSource != kRate`.